File: p2p-ws-sdk-ts/scripts/verify_wrapper.ts
import { loadProtoRoot } from "../src/proto.js"
import { decodeWrapper, encodeWrapper } from "../src/wrapper.js"

const root = await loadProtoRoot()
const plain = encodeWrapper(root, { seq: 1, command: 123, data: new Uint8Array([1, 2, 3]) })
const back = decodeWrapper(root, plain)

if (back.seq !== 1) throw new Error("seq mismatch")
if (back.command !== 123) throw new Error("command mismatch")
if (!back.data || Buffer.from(back.data).toString("hex") !== "010203") throw new Error("data mismatch")

console.log("ok=1")

File: p2p-ws-sdk-python/scripts/verify_wrapper.py
from __future__ import annotations

from pathlib import Path
import sys

root = Path(__file__).resolve().parents[1]
sys.path.insert(0, str(root / "src"))

from p2p_ws_sdk.wrapper import P2PWrapper, decode_wrapper, encode_wrapper


def main() -> int:
    w = P2PWrapper(seq=1, command=123, data=b"\x01\x02\x03")
    data = encode_wrapper(w)
    back = decode_wrapper(data)
    if back.seq != 1:
        raise SystemExit("seq mismatch")
    if back.command != 123:
        raise SystemExit("command mismatch")
    if back.data != b"\x01\x02\x03":
        raise SystemExit("data mismatch")
    print("ok=1")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

